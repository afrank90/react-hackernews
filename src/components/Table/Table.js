import React from 'react';
import Button from '../Button';
import PropTypes from 'prop-types';

/**
 * Table to render list of items.
 */
const Table = ({ list, onDismiss }) => (
  <div className="c-table">
    {list.map(item => (
      <div key={item.objectID} className="c-table__row">
        <span>
          <a href={item.url} className="c-column--large">
            {item.title}{' '}
          </a>
          <span className="c-column--medium">{item.author} </span>
          <span className="c-column--small">{item.comments} </span>
          <span className="c-column--small">{item.points}</span>
        </span>

        <span>
          <Button
            onClick={() => onDismiss(item.objectID)}
            className="c-button--inline"
          >
            Dismiss
          </Button>
        </span>
      </div>
    ))}
  </div>
);

/**
 * Defining properties types.
 * Some kind of equivalent to TypeScript for component properties.
 *
 * In this example we can specify also type for expected array values.
 */
Table.propTypes = {
  list: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      author: PropTypes.string,
      num_comments: PropTypes.number,
      points: PropTypes.number,
      objectID: PropTypes.string.isRequired
    })
  ).isRequired,
  onDismiss: PropTypes.func.isRequired
};

export default Table;
