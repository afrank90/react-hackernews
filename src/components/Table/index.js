import React from 'react';
import Button from '../Button';

/**
 * Table to render list of items.
 */
const Table = ({ list, onDismiss }) => (
  <div className="c-table">
    {list.map(item => (
      <div key={item.objectID} className="c-table__row">
        <span>
          <a href={item.url} className="c-column--large">
            {item.title}{' '}
          </a>
          <span className="c-column--medium">{item.author} </span>
          <span className="c-column--small">{item.comments} </span>
          <span className="c-column--small">{item.points}</span>
        </span>

        <span>
          <Button
            onClick={() => onDismiss(item.objectID)}
            className="c-button--inline"
          >
            Dismiss
          </Button>
        </span>
      </div>
    ))}
  </div>
);

export default Table;
