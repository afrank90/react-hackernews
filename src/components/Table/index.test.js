import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Table from './';

Enzyme.configure({ adapter: new Adapter() });

describe('Table', () => {
  const props = {
    list: [
      {
        title: 'title 1',
        author: 'author 1',
        num_comments: 1,
        points: 2,
        objectID: 123
      },
      {
        title: 'title 2',
        author: 'author 2',
        num_comments: 4,
        points: 3,
        objectID: 1234
      }
    ],
    onDismiss: () => {}
  };

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Table {...props} />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('has valid snapshot', () => {
    const component = renderer.create(<Table {...props} />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('shows two items in the list', () => {
    const element = shallow(<Table {...props} />);

    expect(element.find('.c-table__row')).toHaveLength(2);
  });
});
