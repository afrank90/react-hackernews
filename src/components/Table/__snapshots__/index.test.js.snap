// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table has valid snapshot 1`] = `
<div
  className="c-table"
>
  <div
    className="c-table__row"
  >
    <span>
      <a
        className="c-column--large"
        href={undefined}
      >
        title 1
         
      </a>
      <span
        className="c-column--medium"
      >
        author 1
         
      </span>
      <span
        className="c-column--small"
      >
         
      </span>
      <span
        className="c-column--small"
      >
        2
      </span>
    </span>
    <span>
      <button
        className="c-button--inline"
        onClick={[Function]}
        type="button"
      >
        Dismiss
      </button>
    </span>
  </div>
  <div
    className="c-table__row"
  >
    <span>
      <a
        className="c-column--large"
        href={undefined}
      >
        title 2
         
      </a>
      <span
        className="c-column--medium"
      >
        author 2
         
      </span>
      <span
        className="c-column--small"
      >
         
      </span>
      <span
        className="c-column--small"
      >
        3
      </span>
    </span>
    <span>
      <button
        className="c-button--inline"
        onClick={[Function]}
        type="button"
      >
        Dismiss
      </button>
    </span>
  </div>
</div>
`;
